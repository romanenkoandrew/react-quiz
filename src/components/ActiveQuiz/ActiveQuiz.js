import React from "react";
import styles from "./ActiveQuiz.module.css";
import AnswerList from "./AnswerList/AnswerList";

const ActiveQuiz = (props) => {
  return (
    <div className={styles.ActiveQuiz}>
      <p className={styles.Question}>
        <span>
          <strong>{props.answerNumber}</strong> &nbsp; {props.question}
        </span>
        <small>
          {props.answerNumber} из {props.quizLength}
        </small>
      </p>
      <AnswerList
        answers={props.answers}
        onAnswerClick={props.onAnswerClick}
        state={props.state}
      />
    </div>
  );
};

export default ActiveQuiz;
