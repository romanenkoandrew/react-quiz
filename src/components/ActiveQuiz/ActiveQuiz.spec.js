import React from "react";
import { shallow } from "enzyme";
import ActiveQuiz from "./ActiveQuiz";

describe("ActiveQuiz component", () => {
  it("should render ActiveQuiz with props.state", () => {
    const component = shallow(
      <ActiveQuiz
        answerNumber={1}
        quizLength={3}
        question={"question"}
        state={true}
        answers={[{ answer: { text: "test", id: "test" } }]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
