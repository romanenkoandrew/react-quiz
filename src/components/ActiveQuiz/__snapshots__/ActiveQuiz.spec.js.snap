// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveQuiz component should render ActiveQuiz with props.state 1`] = `
<div
  className="ActiveQuiz"
>
  <p
    className="Question"
  >
    <span>
      <strong>
        1
      </strong>
         
      question
    </span>
    <small>
      1
       из 
      3
    </small>
  </p>
  <AnswerList
    answers={
      Array [
        Object {
          "answer": Object {
            "id": "test",
            "text": "test",
          },
        },
      ]
    }
    state={true}
  />
</div>
`;
